spec/unit/network/http/connection_spec.rb

```diff
@@ -160,12 +160,33 @@ def retry_after(datetime)
     end
 
     it "should return a 503 response if Retry-After is not convertible to an Integer or RFC 2822 Date" do
-      stub_request(:get, url).to_return(status: [503, 'Service Unavailable'], headers: {'Retry-After' => 'foo'})
+      retry_after('foo')
 
       result = subject.get('/foo')
       expect(result.code).to eq("503")
     end
 
+    it "should close the connection before sleeping" do
+      retry_after('42')
+
+      http1 = Net::HTTP.new(host, port)
+      http1.use_ssl = true
+      allow(http1).to receive(:started?).and_return(true)
+
+      http2 = Net::HTTP.new(host, port)
+      http2.use_ssl = true
+      allow(http1).to receive(:started?).and_return(true)
```

Comment:
should this be `http2`?

Reply:
🤦‍♂

Reply:
It's interesting that tests passed with this

Reply:
true dat. Since the expectation is that http1 is finished before we sleep, this line isn't strictly necessary, as it just means the `close_connection` call is a noop when called with http2.